packages/code-du-travail-api/src/server/routes/items/getRelatedItems.js

```diff
@@ -52,12 +121,35 @@ async function getRelatedItems({ title, settings, slug }) {
 
   const { hits: { hits: semanticHits } = { hits: [] } } = semResponse;
   const { hits: { hits: fullTextHits } = { hits: [] } } = esResponse;
-  const [rootSlug] = slug.split("#");
 
-  return utils
-    .mergePipe(fullTextHits, semanticHits, MAX_RESULTS)
-    .filter(({ _source }) => !_source.slug.startsWith(rootSlug))
-    .map(({ _source }) => _source);
+  return (
+    utils
+      .mergePipe(fullTextHits, semanticHits, MAX_RESULTS)
+      // we filter fields and add some info about recommandation type for evaluation purpose
+      .map(({ _source }) => mapSource("search", _source))
+  );
+}
+
+// get related items, depending on : covisits present & non empty and A/B testing ratio
+async function getRelatedItems({ title, settings, slug, covisits }) {
+  const useCovisit = Math.random() < abRatio;
+
+  const covisitedItems =
+    covisits && useCovisit
+      ? await getCovisitedItems({ covisits, slug })
+      : undefined;
```

Comment:
I would return and empty array and then 
```js 
  const relatedItems = covisitedItems.length > 0
    ? covisitedItems
    : await getSearchBasedItems({ settings, slug, title });
```
so we can fallback on previous relatedItem if no covisits